src/curve_fit.jl

```diff
@@ -29,18 +29,25 @@ function lmfit(f, g, p0::AbstractArray, wt::AbstractArray; kwargs...)
 end
 
 #for inplace f and inplace g
-function lmfit(f!, g!, p0::AbstractArray, wt::AbstractArray, r::AbstractArray; kwargs...)
-    R = OnceDifferentiable(f!, g!, p0, similar(r); inplace = true)
+function lmfit(f, g, p0::AbstractArray, wt::AbstractArray, r::AbstractArray; inplace=false, kwargs...)
```

Comment:
It'd probably be nice to actually keep `f!` and `g!`, also we don't need the `inplace=false` anymore